packages/cli/src/lib/snapshot/snapshot.spec.ts

```diff
@@ -1,28 +1,75 @@
 jest.mock('../platform/authenticatedClient');
 jest.mock('fs');
 
-import {createReadStream, ReadStream} from 'fs';
-import {join} from 'path';
-import {Readable} from 'stream';
+import {
+  ResourceSnapshotsModel,
+  ResourceSnapshotsReportModel,
+  ResourceSnapshotsReportResultCode,
+  ResourceSnapshotsReportStatus,
+  ResourceSnapshotsReportType,
+} from '@coveord/platform-client';
 import {mocked} from 'ts-jest/utils';
 import {AuthenticatedClient} from '../platform/authenticatedClient';
-import {SnapshotFactory} from './snapshotFactory';
+import {ISnapshotValidation, Snapshot} from './snapshot';
 
-const mockedCreateReadStream = mocked(createReadStream);
 const mockedAuthenticatedClient = mocked(AuthenticatedClient);
 const mockedCreateSnapshotFromFile = jest.fn();
 const mockedPushSnapshot = jest.fn();
+const mockedGetSnapshot = jest.fn();
+const mockedDryRunSnapshot = jest.fn();
 const mockedGetClient = jest.fn();
 
-const doMockReadStream = () => {
-  mockedCreateReadStream.mockImplementation(() => {
-    const buffer = Buffer.from('this is a tést');
-    const readable = new Readable();
-    readable._read = () => {};
-    readable.push(buffer);
-    readable.push(null);
-    return readable as unknown as ReadStream;
-  });
+const getReport = (
+  snapshotId: string,
+  type: ResourceSnapshotsReportType,
+  status: ResourceSnapshotsReportStatus,
+  resultCode: ResourceSnapshotsReportResultCode
+): ResourceSnapshotsReportModel => {
+  return {
+    id: snapshotId,
+    updatedDate: 1622555847000,
+    type: type,
+    status: status,
+    resultCode: resultCode,
+    resourcesProcessed: 12,
+    resourceOperations: {},
+    resourceOperationResults: {},
+  };
+};
+
+const getSuccessReport = (snapshotId: string): ResourceSnapshotsReportModel => {
+  return getReport(
+    snapshotId,
+    ResourceSnapshotsReportType.DryRun,
+    ResourceSnapshotsReportStatus.Completed,
+    ResourceSnapshotsReportResultCode.Success
+  );
+};
+
+const getErrorReport = (snapshotId: string): ResourceSnapshotsReportModel => {
+  return getReport(
+    snapshotId,
+    ResourceSnapshotsReportType.DryRun,
+    ResourceSnapshotsReportStatus.Completed,
+    ResourceSnapshotsReportResultCode.ResourcesInError
+  );
+};
+
+const getDummySnapshotModel = (
+  orgId: string,
+  snapshotId: string,
+  reports: ResourceSnapshotsReportModel[] = []
+): ResourceSnapshotsModel => {
+  return {
+    id: snapshotId,
+    createdBy: 'user@coveo.com',
+    createdDate: 1622555047116,
+    targetId: orgId,
+    developerNote: 'hello',
+    reports: reports,
+    synchronizationReports: [],
+    contentSummary: {EXTENSION: 1, FIELD: 11},
+  };
 };
```

Comment:
Not necessarly prettier compliant (i.e. might need a alt-shift-f in vscode)
But yeah, useless to do a full block function imo.

```suggestion
): ResourceSnapshotsModel => {
  {
    id: snapshotId,
    createdBy: 'user@coveo.com',
    createdDate: 1622555047116,
    targetId: orgId,
    developerNote: 'hello',
    reports: reports,
    synchronizationReports: [],
    contentSummary: {EXTENSION: 1, FIELD: 11},
  };
```